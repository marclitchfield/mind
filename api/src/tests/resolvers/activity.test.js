import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Activity tests', () => {
  applySetup();

  test('create activity for ability', () => {
    return verifyMutations('Activity tests', 'create activity for ability', {
      mutations: [
        ({space}) => gql`mutation { Ability { ability:post_in_space(input:{sourceId:"${space.id}", title:"test ability"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title:"test event", datetime:"2011-12-12"}) { id } } }`,
        ({event, ability}) => gql`mutation { Activity { activity:post_for_ability(input:{sourceId:"${ability.id}", eventId:"${event.id}", title:"test ability activity"}) { id } } }`
      ],
      query: ({ability}) => gql`query {
        Ability(id:"${ability.id}") {
          title
          activities {
            title
            ability { title }
            event { title }
          }
        }
      }`
    });
  });

  test('create ability activity for person', () => {
    return verifyMutations('Activity tests', 'create ability activity for person', {
      mutations: [
        ({space}) => gql`mutation { Ability { ability:post_in_space(input:{sourceId:"${space.id}", title:"test ability"}) { id } } }`,
        ({space}) => gql`mutation { Person { person:post_in_space(input:{sourceId:"${space.id}", title:"test person"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title:"test event", datetime:"2011-12-12"}) { id } } }`,
        ({ability, person, event}) => gql`mutation { Activity { post_for_ability(input:{sourceId:"${ability.id}", personId:"${person.id}", eventId:"${event.id}" title:"test ability activity"}) { id } } }`
      ],
      query: ({ability}) => gql`query {
        Ability(id:"${ability.id}") {
          title
          activities {
            title
            ability { title }
            event { title }
            person { 
              title
              abilities {
                title
              }
              activities {
                title
              }
            }
          }
        }
      }`
    });
  });

  test('create objective activity for person', () => {
    return verifyMutations('Activity tests', 'create objective activity for person', {
      mutations: [
        ({space}) => gql`mutation { Objective { objective:post_in_space(input:{sourceId:"${space.id}", title:"test objective"}) { id } } }`,
        ({space}) => gql`mutation { Person { person:post_in_space(input:{sourceId:"${space.id}", title:"test person"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title:"test event", datetime:"2011-12-12"}) { id } } }`,
        ({objective, person, event}) => gql`mutation { Activity { post_for_objective(input:{sourceId:"${objective.id}", personId:"${person.id}", eventId:"${event.id}" title:"test objective activity"}) { id } } }`
      ],
      query: ({objective}) => gql`query {
        Objective(id:"${objective.id}") {
          title
          activities {
            title
            objective { title }
            event { title }
            person { 
              title
              objectives {
                title
              }
              activities {
                title
              }
            }
          }
        }
      }`
    });
  });

  test('create ability activity for item', () => {
    return verifyMutations('Activity tests', 'create ability activity for item', {
      mutations: [
        ({space}) => gql`mutation { Ability { ability:post_in_space(input:{sourceId:"${space.id}", title:"test ability"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title:"test event", datetime:"2011-12-12"}) { id } } }`,
        ({space}) => gql`mutation { Concept { type:post_in_space(input:{sourceId:"${space.id}", title:"test concept"}) { id } } }`,
        ({space, type}) => gql`mutation { Item { item:post_in_space(input:{sourceId:"${space.id}", classId:"${type.id}", title:"test item"}) { id } } }`,
        ({ability, item, event}) => gql`mutation { Activity { post_for_ability(input:{sourceId:"${ability.id}", itemId:"${item.id}", eventId:"${event.id}" title:"test ability activity"}) { id } } }`
      ],
      query: ({ability}) => gql`query {
        Ability(id:"${ability.id}") {
          title
          activities {
            title
            ability { title }
            event { title }
            item { 
              title
              abilities {
                title
              }
              activities {
                title
              }
            }
          }
        }
      }`
    });
  });

  test('remove ability activity', () => {
    return verifyMutations('Activity tests', 'remove ability activity', {
      mutations: [
        ({space}) => gql`mutation { Ability { ability:post_in_space(input:{sourceId:"${space.id}", title:"test ability"}) { id } } }`,
        ({space}) => gql`mutation { Person { person:post_in_space(input:{sourceId:"${space.id}", title:"test person"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title:"test event", datetime:"2011-12-12"}) { id } } }`,
        ({ability, person, event}) => gql`mutation { Activity { activity:post_for_ability(input:{sourceId:"${ability.id}", personId:"${person.id}", eventId:"${event.id}" title:"test ability activity"}) { id } } }`,
        ({ability, person, event, activity}) => gql`mutation { Activity { post_for_ability(input:{remove:true, id:"${activity.id}", sourceId:"${ability.id}", personId:"${person.id}", eventId:"${event.id}"}) { id } } }`
      ],
      query: ({space}) => gql`query {
        Space(id:"${space.id}") {
          title
          abilities { 
            title 
            activities {
              title
              ability { title }
              event { title }
              person { title }
            }
          }
          activities { title }
        }
      }`
    });
  });

});
