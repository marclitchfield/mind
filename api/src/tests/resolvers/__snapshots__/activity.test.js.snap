// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity tests create ability activity for item 1`] = `
Object {
  "Ability": Object {
    "__typename": "Ability",
    "activities": Array [
      Object {
        "__typename": "Activity",
        "ability": Object {
          "__typename": "Ability",
          "title": "test ability",
        },
        "event": Object {
          "__typename": "Event",
          "title": "test event",
        },
        "item": Object {
          "__typename": "Item",
          "abilities": Array [
            Object {
              "__typename": "Ability",
              "title": "test ability",
            },
          ],
          "activities": Array [
            Object {
              "__typename": "Activity",
              "title": "test ability activity",
            },
          ],
          "title": "test item",
        },
        "title": "test ability activity",
      },
    ],
    "title": "test ability",
  },
}
`;

exports[`Activity tests create ability activity for person 1`] = `
Object {
  "Ability": Object {
    "__typename": "Ability",
    "activities": Array [
      Object {
        "__typename": "Activity",
        "ability": Object {
          "__typename": "Ability",
          "title": "test ability",
        },
        "event": Object {
          "__typename": "Event",
          "title": "test event",
        },
        "person": Object {
          "__typename": "Person",
          "abilities": Array [
            Object {
              "__typename": "Ability",
              "title": "test ability",
            },
          ],
          "activities": Array [
            Object {
              "__typename": "Activity",
              "title": "test ability activity",
            },
          ],
          "title": "test person",
        },
        "title": "test ability activity",
      },
    ],
    "title": "test ability",
  },
}
`;

exports[`Activity tests create activity for ability 1`] = `
Object {
  "Ability": Object {
    "__typename": "Ability",
    "activities": Array [
      Object {
        "__typename": "Activity",
        "ability": Object {
          "__typename": "Ability",
          "title": "test ability",
        },
        "event": Object {
          "__typename": "Event",
          "title": "test event",
        },
        "title": "test ability activity",
      },
    ],
    "title": "test ability",
  },
}
`;

exports[`Activity tests create objective activity for person 1`] = `
Object {
  "Objective": Object {
    "__typename": "Objective",
    "activities": Array [
      Object {
        "__typename": "Activity",
        "event": Object {
          "__typename": "Event",
          "title": "test event",
        },
        "objective": Object {
          "__typename": "Objective",
          "title": "test objective",
        },
        "person": Object {
          "__typename": "Person",
          "activities": Array [
            Object {
              "__typename": "Activity",
              "title": "test objective activity",
            },
          ],
          "objectives": Array [
            Object {
              "__typename": "Objective",
              "title": "test objective",
            },
          ],
          "title": "test person",
        },
        "title": "test objective activity",
      },
    ],
    "title": "test objective",
  },
}
`;

exports[`Activity tests remove ability activity 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "abilities": Array [
      Object {
        "__typename": "Ability",
        "activities": Array [],
        "title": "test ability",
      },
    ],
    "activities": Array [],
    "title": "remove ability activity",
  },
}
`;
